pkcs8/src/one_asymmetric_key.rs

```diff
@@ -0,0 +1,143 @@
+use der::{BitString, ContextualTo1, Decodable, Encodable, Message};
+
+use crate::{
+    common::{Version, _AttributesStub},
+    AlgorithmIdentifier, Error, Result,
+};
+use core::{convert::TryFrom, fmt};
+
+/// PKCS#8 `OneAsymmetricKey`.
+///
+/// ASN.1 structure containing a [`Version`], an [`AlgorithmIdentifier`], private key
+/// data, and optionally public key data, in an algorithm specific format.
+///
+/// Described in [RFC 5958 Section 2]:
+///
+/// ```text
+/// OneAsymmetricKey ::= SEQUENCE {
+///     version                   Version,
+///     privateKeyAlgorithm       PrivateKeyAlgorithmIdentifier,
+///     privateKey                PrivateKey,
+///     attributes            [0] Attributes OPTIONAL,
+///     ...,
+///     [[2: publicKey        [1] PublicKey OPTIONAL ]],
+///     ...
+///   }
+///
+/// Version ::= INTEGER { v1(0), v2(1) } (v1, ..., v2)
+///
+/// PrivateKeyAlgorithmIdentifier ::= AlgorithmIdentifier
+///
+/// PrivateKey ::= OCTET STRING
+///
+/// Attributes ::= SET OF Attribute
+///
+/// PublicKey ::= BIT STRING
+/// ```
+///
+/// [RFC 5958 Section 2]: https://datatracker.ietf.org/doc/html/rfc5958#section-2
+#[derive(Clone)]
+pub struct OneAsymmetricKey<'a> {
+    /// Version, can be set to [`Version::V1`] or [`Version::V2`]
```

Comment:
Is "can be set to …" not kind of redundant with the type being an enum, and prone to becoming outdated?

Reply:
Good point, lemme change that